dll/win32/uxtheme/png.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * PROJECT:     ReactOS uxtheme.dll
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     PNG image support in visual styles
+ * COPYRIGHT:   Copyright 2023 Ethan Rodensky <Splitwirez@gmail.com>
+ */
+
+#include "uxthemep.h"
+
+#ifdef MSSTYLES_ACCEPT_PNG_IMAGES
+#include <ole2.h>
+#include <objidl.h>
+#include <gdiplus.h>
+#include <gdipluscolor.h>
+#include <winreg.h>
+#include <shlwapi.h>
+
+BOOL
+MSSTYLES_TryLoadPng(
+    _In_ HINSTANCE hTheme,
+    _In_ LPCWSTR szFile,
+    _In_ LPCWSTR type,
+    _Out_ HBITMAP *pngBitmap)
+{
+    BOOL isPNG = FALSE;
+
+    HRSRC pngRes = FindResourceW(hTheme, szFile, type);
+    HGLOBAL pngAlloc = LoadResource(hTheme, pngRes);
+    DWORD pngSize = SizeofResource(hTheme, pngRes);
+    LPVOID pngLock = LockResource(pngAlloc);
+    
+    if (!pngLock)
+        return FALSE;
+    
+    IStream* pngStream = SHCreateMemStream((BYTE*)pngLock, pngSize);
+    if (!pngStream)
```

Comment:
I suppose that if you remove the explicit `(BYTE*)` case you get a compilation error?
```suggestion
    IStream* pngStream = SHCreateMemStream((BYTE*)pData, dwSize);
    if (!pngStream)
```